motr/setup.c

```diff
@@ -2484,7 +2484,18 @@ static int cs_level_enter(struct m0_module *module)
 	int                     rc;
 
 	M0_ENTRY("cctx=%p level=%d level_name=%s", cctx, level, level_name);
-	if (gotsignal)
+	/** If a signal is recieved in CS_LEVEL_REQH_STOP_WORKAROUND,
+	  * m0_module_fini is called, In m0_module_fini
+	  * CS_LEVEL_RPC_MACHINES_INIT would assert because there are
+	  * active ongoing connection.
+	  * ideally connections are closed in cs_ha_fini during
+	  * cs_level_leave:: CS_LEVEL_REQH_STOP_WORKAROUND.
+	  * In cs_level_enter::CS_LEVEL_REQH_STOP_WORKAROUND, it does nothing
+	  * so just skip the error in CS_LEVEL_REQH_STOP_WORKAROUND
+	  * which will be handled in the next state. during fini, cs_ha_fini
+	  * is called before CS_LEVEL_RPC_MACHINES_INIT
+	 */
+	if (gotsignal && level != CS_LEVEL_REQH_STOP_WORKAROUND)
```

Comment:
[Suggestion]
Would be good if we add some comment in case: CS_LEVEL_REQUEST_STOP_WORKAROUND.
mentioning "Check comment and revisit this logic whenever you add some code here"
Because this logic might need to revisit in such case and this comment might overlooked 
at that time since not at same place.